### ISSUE TITLE
[Bug] p-calendar 12 hour format doesn't reflect the time selected

### ISSUE THREAD

**Issue by luchillo17** [primefaces/primeng#2539]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
```

**Plunkr Case (Bug Reports)**
Please fork the plunkr below and create a case demonstrating your bug report. Issues without a plunkr have much less possibility to be reviewed.

**Current behavior**
Using 12 time format will render always `AM` in the date input, even if the actual time when the popup shows is in `PM`, it's just a graphical issue but is very missleading, the actual date in the form is correct.

**Expected behavior**

**Minimal reproduction of the problem with instructions**
Maybe just adding a '12 hour' example to the showcase to see if that happens the same.

**What is the motivation / use case for changing the behavior?**
Miss-leading behavior affects UX and makes easy to make errors in the program.

* **Angular version:** 4.0.2

* **PrimeNG version:** 4.0.0-rc.3

* **Browser:** [all | Chrome 57]

 
* **Language:** [all | TypeScript 2.3.0]

* **Extra info:**
Check this image, the input says `AM`, the dropdown says `PM` and the form returns the actual date in 24 format with the expected `PM` time to 24 hours format:
![image](https://cloud.githubusercontent.com/assets/9698639/25092081/bce47de2-2351-11e7-9e02-5d1ce3e767de.png)

**Comment by kc13greiner** [comment]
I am having the same issue:

Angular version: 4.0.2
PrimeNG version: 4.0.0-rc.3
Browser: [all | Chrome 57]

Example:

Selection:
![image](https://cloud.githubusercontent.com/assets/21210601/25250431/96e7b660-25e3-11e7-921b-3721580e900f.png)

Output From Above selection:
![image](https://cloud.githubusercontent.com/assets/21210601/25250449/a30d1f84-25e3-11e7-9802-eca45fa5911f.png)

Weird 12 o'clock Selection:
![image](https://cloud.githubusercontent.com/assets/21210601/25250519/dc868c78-25e3-11e7-8d55-960896159159.png)

![image](https://cloud.githubusercontent.com/assets/21210601/25250528/e75a652a-25e3-11e7-952c-9276673c47bb.png)

**Comment by Tuanpham007** [comment]
There is a bug with the formatTime function in calendar.ts (line 991). 

On line 1014,

`
if(this.hourFormat == '12') {
            output += hours > 11 ? ' PM' : ' AM';
        }`

the statement checks if hours is greater than 11 if so, it sets the output to "PM" otherwise "AM". However, hours will not be greater than 11 in most causes because on line 1001 

`if(this.hourFormat == '12' && hours > 11 && hours != 12) {
            hours-=12;
        }`

For instance, if you selected 5:00PM (17:00), 17-12 = 5... the hours variable = 5 which is not greater than 11 therefore the output displays "AM" instead of "PM".

Here's the code for a fix:

`formatTime(date) {
        if(!date) {
            return '';
        }
        
        let output = '';
        let origHours = date.getHours();
        let hours = date.getHours();
        let minutes = date.getMinutes();
        let seconds = date.getSeconds();
        
        if(this.hourFormat == '12' && hours > 11 && hours != 12) {
            hours-=12;
        }
        
        output += (hours < 10) ? '0' + hours : hours;
        output += ':';
        output += (minutes < 10) ? '0' + minutes : minutes;
        
        if(this.showSeconds) {
            output += ':';
            output += (seconds < 10) ? '0' + seconds : seconds;
        }
        
        if(this.hourFormat == '12') {
            output += origHours > 11 ? ' PM' : ' AM';
        }
        
        return output;
    }
`

**Comment by luchillo17** [comment]
What's the reason for `hours > 11 && hours != 12`?

Isn't that effectively the same as `hours > 12`?

**Comment by Tuanpham007** [comment]
Yes, I'd say so. That was from the original code which I did not change. They changes I've added are where "origHours" appear.

**Comment by luchillo17** [comment]
I'd say it's not a big performance difference to compress as i've said, but it makes the code more readable, it's actually confusing to look at that conditional, it leaves me wondering whyyyy?

**Comment by khyatiaggarwal** [comment]
Same here, issue still present. I just updated and it is still not displaying correct time format.
Actually it read time of system watch and according system watch time, it set AM/PM. Actually Boolean 'pm' is set true/false according to system watch. And its value is not getting updated according to calendar value. Now I have set condition in calender.js file in formatTime() function, its working fine. For reference:- 
Calendar.prototype.formatTime = function (date) {
        if (!date) {
            return '';
        }
        var output = '';
        var hours = date.getHours();
        var minutes = date.getMinutes();
        var seconds = date.getSeconds();
        if (this.hourFormat == '12' && hours > 11 && hours != 12) {
            hours -= 12;
        }
        output += (hours < 10) ? '0' + hours : hours;
        output += ':';
        output += (minutes < 10) ? '0' + minutes : minutes;
        if (this.showSeconds) {
            output += ':';
            output += (seconds < 10) ? '0' + seconds : seconds;
        }
        if (this.hourFormat == '12') {
            output += date.getHours() > 11 ? ' PM' : ' AM';
		

# ##  **if(date.getHours() <= 11)
				this.pm = false;**

	
        }
		
        return output;
    };

Please change code in calender.js file.

**Comment by nitishGE** [comment]
I am facing the same issue. At output += date.getHours() > 11 ? ' PM' : ' AM'; why are we not considering the user selected AM/PM value. Rather we are taking it from system time. This is where the issue comes.
